src/Analyzers/Core/CodeFixes/UseNullPropagation/AbstractUseNullPropagationCodeFixProvider.cs

```diff
@@ -153,6 +156,24 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
                 syntaxFacts, generator, whenPartIsNullable, whenTrueStatement, match);
             Contract.ThrowIfNull(newWhenTrueStatement);
 
+            // If we have code like:
+            // ...
+            // else if (v != null)
+            // {
+            //     v.M();
+            // }
+            // then we want to keep the result statement in a block:
+            // else
+            // {
+            //     v?.M();
+            // }
+            // Applies only to C# since VB doesn't have a general-purpose block syntax
+            if (syntaxFacts.IsElseClause(originalIfStatement.Parent) &&
+                IsBlockSyntax(whenTrueStatement.Parent))
+            {
+                newWhenTrueStatement = Block(newWhenTrueStatement).WithAdditionalAnnotations(Formatter.Annotation);
```

Comment:
Instead of making a new block syntax, you could just reuse the existing one, but replace the single statement within with the new statement you're generating.